### PR TITLE
[Ruler] Add `cortex_prometheus_last_evaluation_samples` metric

### PR DESCRIPTION
**Pull request by gotjosh**
**What this PR does**:

Useful to know how many samples a given rule group is producing. Based on the upstream done in https://github.com/prometheus/prometheus/pull/7977

**Checklist**
- [x] Tests updated
- [x] Documentation added
- [x] `CHANGELOG.md` updated - the order of entries should be `[CHANGE]`, `[FEATURE]`, `[ENHANCEMENT]`, `[BUGFIX]`

### CODE REVIEW

File: pkg/ruler/ruler_ring.go [hunk 1]
```diff
@@ -63,7 +63,7 @@ func (cfg *RingConfig) RegisterFlags(f *flag.FlagSet) {
 	f.StringVar(&cfg.InstanceAddr, "ruler.ring.instance-addr", "", "IP address to advertise in the ring.")
 	f.IntVar(&cfg.InstancePort, "ruler.ring.instance-port", 0, "Port to advertise in the ring (defaults to server.grpc-listen-port).")
 	f.StringVar(&cfg.InstanceID, "ruler.ring.instance-id", hostname, "Instance ID to register in the ring.")
-	f.IntVar(&cfg.NumTokens, "ruler.ring.num-tokens", 128, "Number of tokens for each ingester.")
```
Comment (gotjosh) [inline pkg/ruler/ruler_ring.go:66]: unrelated but a tiny typo I found while browsing.

**Review by pstibrany — APPROVED** [review]
LGTM, thanks!

File: CHANGELOG.md [hunk 2]
```diff
@@ -2,6 +2,7 @@
 
 ## master / unreleased
 
+* [ENHANCEMENT] Ruler: Addd `cortex_prometheus_last_evaluation_samples` to expose the number of samples generated by a rule group per tenant. #3582
```
Comment (pstibrany) [inline CHANGELOG.md:5]: Typo in Addd

**Review by pracucci — APPROVED** [review]
Changes LGTM. I'm a bit worried about the cardinality of this metric, then I noticed we already have other 6 metrics with the same cardinality, so 1 more will not be an issue.

**Pull request merged**